In [40]:
import pandas as pd
from ingredients_lists import *

In [2]:
data = pd.read_csv('ingredientes_estandarizados.csv')

In [3]:
df = data.copy()

In [4]:
df

receta  \
0                  Iogurt grec amb fraules   
1                  Iogurt grec amb fraules   
2                  Iogurt grec amb fraules   
3                  Iogurt grec amb fraules   
4                  Iogurt grec amb fraules   
...                                    ...   
1058  Pollastre al forn a les fines herbes   
1059  Pollastre al forn a les fines herbes   
1060  Pollastre al forn a les fines herbes   
1061  Pollastre al forn a les fines herbes   
1062          Broqueta de gambes amb xoriç   

                    ingrediente_original  cantidad unidad ingrediente_estandar  
0                       Iogurt grec 200g       200      g          iogurt grec  
1                       Fraules 100-150g       150      g              fraules  
2                    Nous i ametlles 30g        30      g      nous i ametlles  
3                             Xía 10-20g        20      g                  xía  
4                       Canyella al gust         0    NaN             canyella  
...                                  ...       ...    ...                  ...  
1058    5g d'orenga fresc (0.5g en pols)         0    NaN         orenga fresc  
1059               15g de mostassa Dijon         0    NaN       mostassa dijon  
1060  15ml d'oli d'oliva (una cullerada)         0    NaN          oli d'oliva  
1061                 Sal i pebre al gust         0    NaN          sal i pebre  
1062                          llagostins       250      g           llagostins  

[1063 rows x 5 columns]

In [5]:
df.columns

Index(['receta', 'ingrediente_original', 'cantidad', 'unidad',
       'ingrediente_estandar'],
      dtype='object')

In [6]:
df.shape

(1063, 5)

In [7]:
df['ingrediente_estandar'].nunique()

231

### Líneas para manipular!

In [63]:
for index, row in df.iterrows():
    if row['ingrediente_estandar'] in condiments:
        df.at[index, 'tipo_ingrediente'] = 'Condiments'
    else:
        df.at[index, 'tipo_ingrediente'] = row['tipo_ingrediente']

In [68]:
df['tipo_ingrediente'].value_counts()

tipo_ingrediente
Condiments         317
Verdures           275
Làctics i ous      169
Altres             151
Fruites             55
Carns vermellas     38
Peixos              27
Carns blanques      17
Mariscos            11
Name: count, dtype: int64

In [66]:
df.sort_values(by=['ingrediente_estandar', 'cantidad'], ascending=[True, True])

receta  \
14   Solomillo de vedella Strogonoff i arròs basmati   
27                                Cuinat de llenties   
74                                         Vinagreta   
272  Ous bullits amb salmó fumat, tàperes i maionesa   
608                                        Pa de lli   
..                                               ...   
603                                        Pa de lli   
672                                      Cetogranola   
3                            Iogurt grec amb fraules   
733                Iogurt grec amb ametlles i nabius   
898    Pudding de xía amb fruits vermells i ametlles   

            ingrediente_original  cantidad unidad ingrediente_estandar  \
14                      Agua 10g         0    NaN                aigua   
27                 750ml d'aigua         0    NaN                aigua   
74                 10 ml d'aigua         0    NaN                aigua   
272                        Aigua         0    NaN                aigua   
608        120 g d’aigua calenta         0    NaN                aigua   
..                           ...       ...    ...                  ...   
603        xia (una culleradeta)        10      g                  xía   
672     5-10g de chía (opcional)        10      g                  xía   
3                     Xía 10-20g        20      g                  xía   
733               10-20g de chía        20      g                  xía   
898  3 cullerades soperes de xía        30      g                  xía   

    tipo_ingrediente  
14            Altres  
27            Altres  
74            Altres  
272           Altres  
608           Altres  
..               ...  
603           Altres  
672           Altres  
3             Altres  
733           Altres  
898           Altres  

[1063 rows x 6 columns]

In [39]:
df[df['ingrediente_estandar'].str.contains("pasta", na=False)]

receta  \
68                                        Amanida tofu   
114             Crema de pastanaga amb coco i gingebre   
373             Amanida thai de tofu (formatge tendre)   
549                          Pasta de carabassí scampi   
582                  Crema de pastanaga coco, gingebre   
652                  Verdures bullides amb ous bullits   
726                                  Amanida de colors   
834  Pitreres de pollastre amb verduretes (al forn)...   
850  Pasta de carabassí amb bròquil feta i alfabagu...   
903                                 Estofat de vedella   
941                                     Pastís de carn   

                          ingrediente_original  cantidad unidad  \
68                          100 g de pastanaga         1    NaN   
114                       660 g de pastanagues         6    NaN   
373                          100g de pastanaga         1    NaN   
549  400g de pasta de carbassó (veure recepta)       400      g   
582          600 g de pastanagues (5 mitjanes)         5    NaN   
652                             100g pastanaga         1    NaN   
726                          100g de pastanaga         1    NaN   
834                                1 pastanaga         1    NaN   
850                300 g de pasta de carabassí       300      g   
903                                1 pastanaga         1    NaN   
941                                1 pastanaga         1    NaN   

    ingrediente_estandar tipo_ingrediente  
68             pastanaga             None  
114            pastanaga             None  
373            pastanaga             None  
549   pasta de carabassí             None  
582            pastanaga             None  
652            pastanaga             None  
726            pastanaga             None  
834            pastanaga             None  
850   pasta de carabassí             None  
903            pastanaga             None  
941            pastanaga             None

In [8]:
mask = df[(df['ingrediente_estandar'].str.contains("dents", na=False)) & (df['cantidad'] == 3)]

In [34]:
mask = df[df['ingrediente_estandar'].str.contains("bac", na=False)]

In [36]:
#df.loc[mask.index, 'ingrediente_estandar'] = 'bacallà'
#df.loc[mask.index, 'cantidad'] = None
#df.loc[mask.index, 'unidad'] = None

In [67]:
df.loc[[186],'tipo_ingrediente'] = 'Altres'
#df.loc[[645],'cantidad'] = 1
#df.loc[[781],'unidad'] = None

n_fila = [{
    'receta': 'Broqueta de gambes amb xoriç',
    'ingrediente_original': 'llagostins',
    'cantidad': 250,
    'unidad': 'g',
    'ingrediente_estandar': 'llagostins'
}]

df = pd.concat([df, pd.DataFrame(n_fila)], ignore_index=True)

In [14]:
#df['ingrediente_estandar'] = df['ingrediente_estandar'].replace('salmo fresc', 'salmó fresc')
#df['ingrediente_estandar'] = df['ingrediente_estandar'].replace('romani fresc', 'romaní fresc')


In [69]:
df.to_csv('ingredientes_estandarizados.csv', index=False)